File: voice-bridge/mount_voice.py
#!/usr/bin/env python3
# 把direct_test.wav挂到最新一条assistant消息上（chats集合版）
import re
from pymongo import MongoClient

uri = None
with open('/home/ubuntu/ai-companion/.env') as f:
    for line in f:
        line = line.strip()
        if line.startswith('DATABASE_URL='):
            uri = line.split('=', 1)[1]

client = MongoClient(uri, serverSelectionTimeoutMS=5000)
db = client.get_default_database()
msgs = db['chats']  # 真正的聊天集合

# 找有voice字段的消息，学习格式
ref = msgs.find_one({'role': 'assistant', 'voice': {'$exists': True, '$ne': None}}, sort=[('timestamp', -1)])
if ref:
    print('参考voice格式:', ref.get('voice'))
    v = ref['voice']
    if isinstance(v, str) and '.wav' in v:
        new_v = re.sub(r'[^/]+\.wav$', 'direct_test.wav', v)
    else:
        new_v = '/voices/direct_test.wav'
else:
    new_v = '/voices/direct_test.wav'
    print('没有参考，用默认格式')

# 挂到最新一条assistant消息
latest = msgs.find_one({'role': 'assistant'}, sort=[('timestamp', -1)])
if latest:
    print('目标消息:', latest['content'][:30])
    msgs.update_one({'_id': latest['_id']}, {'$set': {'voice': new_v}})
    print('已挂载:', new_v)
else:
    print('没找到assistant消息')

File: voice-bridge/mount_voice.js
// 用mongoose把direct_test.wav挂到最新assistant消息
const mongoose = require('/home/ubuntu/ai-companion/node_modules/mongoose');
const fs = require('fs');

// 读.env
const env = fs.readFileSync('/home/ubuntu/ai-companion/.env', 'utf8');
const uri = env.split('\n').find(l => l.startsWith('DATABASE_URL=')).split('=').slice(1).join('=');

mongoose.connect(uri).then(async () => {
  const db = mongoose.connection.db;
  const msgs = db.collection('messages');

  // 找有voice字段的消息，学习格式
  const ref = await msgs.findOne({ role: 'assistant', voice: { $exists: true, $ne: null } }, { sort: { timestamp: -1 } });
  let newV = '/voices/direct_test.wav';
  if (ref && ref.voice) {
    console.log('参考voice格式:', JSON.stringify(ref.voice));
    if (typeof ref.voice === 'string' && ref.voice.includes('.wav')) {
      newV = ref.voice.replace(/[^/]+\.wav$/, 'direct_test.wav');
    }
  } else {
    console.log('没有参考格式，用默认:', newV);
  }

  // 挂到最新一条assistant消息
  const latest = await msgs.findOne({ role: 'assistant' }, { sort: { timestamp: -1 } });
  if (latest) {
    console.log('目标消息:', latest.content.substring(0, 30));
    await msgs.updateOne({ _id: latest._id }, { $set: { voice: newV } });
    console.log('已挂载:', newV);
  } else {
    console.log('没找到assistant消息');
  }
  await mongoose.disconnect();
  process.exit(0);
}).catch(e => { console.error('ERR:', e.message); process.exit(1); });
